### PR TITLE
feature/#106 Support Sqlalchemy 1.4 part 2

### LINKED ISSUE #106: Support Sqlalchemy 1.4
sqlalchemy_exasol python package is not compatible with new version of sqlalchemy 1.4+ when used with turbodbc dialect. Below is the code to reproduce this issue.

```python

from sqlalchemy import create_engine
 
exa_engine = create_engine("exa+turbodbc://sys:exasol@exasoldb:8888")
exa_engine.connect()
 
 
################################### ERROR #########################################
 
 
    exec(code, run_globals)
  File "/app/test.py", line 6, in <module>
    exa_engine.connect()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 3166, in connect
    return self._connection_cls(self, close_with_result=close_with_result)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
    else engine.raw_connection()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 3245, in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 3212, in _wrap_pool_connect
    return fn()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 301, in connect
    return _ConnectionFairy._checkout(self)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 761, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 419, in checkout
    rec = pool._do_get()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/impl.py", line 145, in _do_get
    self._dec_overflow()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/impl.py", line 142, in _do_get
    return self._create_connection()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 247, in _create_connection
    return _ConnectionRecord(self)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 362, in __init__
    self.__connect()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 616, in __connect
    pool.dispatch.connect.for_modify(
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/event/attr.py", line 329, in _exec_w_sync_on_first_run
    self(*args, **kw)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/event/attr.py", line 343, in __call__
    fn(*args, **kw)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 1676, in go
    return once_fn(*arg, **kw)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/create.py", line 675, in first_connect
    dialect.initialize(c)
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 394, in initialize
    self.server_version_info = self._get_server_version_info(
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy_exasol/turbodbc.py", line 82, in _get_server_version_info
    result = connection.execute(query).fetchone()[0].split('.')
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1247, in execute
    return self._exec_driver_sql(
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1546, in _exec_driver_sql
    ret = self._execute_context(
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1696, in _execute_context
    context.pre_exec()
  File "/root/venv/lib/python3.8/site-packages/sqlalchemy_exasol/base.py", line 336, in pre_exec
    if self.isdelete or self.root_connection.dialect.server_version_info < (4, 1, 8):
TypeError: '<' not supported between instances of 'NoneType' and 'tuple'

```

### PR DESCRIPTION
**Pull request by Nicoretti**
- Support-Sqlalchemy 1.4 part-1
- Disable unimplemented sql_exprssion_limit_offset tests

### CODE REVIEW

File: test/integration/sqlalchemy/test_suite.py [hunk 1]
```diff
@@ -2,18 +2,132 @@
...
+        * https://peps.python.org/pep-0249/#executemany
+        """
+    )
+
+    TURBODBC_RATIONAL = cleandoc(
+        """
+        The currently used turbodbc driver returns invalid results.
+        Attention: 
+        * turbodbc maintenance is paused until if it is clear if there is still demand for it
+        * If this test(s) will succeed in the future consider repining the turbodbc driver
```
Comment (redcatbear) [inline test/integration/sqlalchemy/test_suite.py]:
```suggestion
        * If these tests will succeed in the future consider repining the turbodbc driver
```

File: test/integration/sqlalchemy/test_suite.py [hunk 2]
```diff
@@ -2,18 +2,132 @@
...
+        This also is a valid behaviour according to the python DBAPI specification.
+        For more details see also:
+        * https://peps.python.org/pep-0249/ 
+        * https://peps.python.org/pep-0249/#rowcount
+        * https://peps.python.org/pep-0249/#id21
+        * https://peps.python.org/pep-0249/#executemany
+        """
+    )
+
+    TURBODBC_RATIONAL = cleandoc(
```
Comment (redcatbear) [inline test/integration/sqlalchemy/test_suite.py]:
```suggestion
    TURBODBC_RATIONALE = cleandoc(
```
Also change references accordingly.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: test/integration/sqlalchemy/test_suite.py (hunk 1 above)
Comment (Nicoretti) [reply test/integration/sqlalchemy/test_suite.py]: fixed

File: test/integration/sqlalchemy/test_suite.py (hunk 2 above)
Comment (Nicoretti) [reply test/integration/sqlalchemy/test_suite.py]: fixed

**Review by redcatbear — APPROVED** [review]

**Pull request merged**